lib/Targets/AIEVecToCpp/TranslateAIEVecToCpp.cpp

```diff
@@ -1041,6 +1041,37 @@ static LogicalResult printOperation(CppEmitter &emitter,
 // Generate the shuffle intrinsic
 static LogicalResult printOperation(CppEmitter &emitter,
                                     aievec::ShuffleOp shuffleOp) {
+  Value lhs = shuffleOp.getLhs();
+  Value rhs = shuffleOp.getRhs();
+  aievec::ShuffleMode mode = shuffleOp.getMode();
+
+  raw_indented_ostream &os = emitter.ostream();
+
+  // Generate the initialization for the result
+  if (failed(emitter.emitAssignPrefix(*shuffleOp)))
+    return failure();
+
+  os << "shuffle";
+  os << "(";
+  if (!emitter.hasValueInScope(lhs))
+    return failure();
+  os << emitter.getOrCreateName(lhs);
+  os << ", ";
+  if (rhs) {
+    if (!emitter.hasValueInScope(rhs))
+      return failure();
+    os << emitter.getOrCreateName(rhs);
+    os << ", ";
+  }
+  os << "eShuffleMode::shuffle_T" << &stringifyEnum(mode).data()[1];
```

Comment:
Not sure if we can do `substr(1)` here, which is easier to understand...

Reply:
Yup! You caught me being lazy 😂